prom-keycloak-proxy/overlays/nerc-ocp-obs/deployments/prom-keycloak-proxy/deployment.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: prom-keycloak-proxy
+    deployment: prom-keycloak-proxy
+  name: prom-keycloak-proxy
+  namespace: prom-keycloak-proxy
+spec:
+  replicas: 1
+  revisionHistoryLimit: 0
+  selector:
+    matchLabels:
+      app: prom-keycloak-proxy
+      deployment: prom-keycloak-proxy
+  strategy:
+    activeDeadlineSeconds: 21600
+    recreateParams:
+      timeoutSeconds: 600
+    resources: {}
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: prom-keycloak-proxy
+        deployment: prom-keycloak-proxy
+      name: prom-keycloak-proxy
+    spec:
+      containers:
+        - env:
+            - name: PROXY_AUTH_CLIENT_ID
+              valueFrom:
+                secretKeyRef:
+                  key: PROXY_AUTH_CLIENT_ID
+                  name: prom-keycloak-proxy-secrets
+            - name: PROXY_AUTH_CLIENT_SECRET
+              valueFrom:
+                secretKeyRef:
+                  key: PROXY_AUTH_CLIENT_SECRET
+                  name: prom-keycloak-proxy-secrets
+            - name: PROXY_AUTH_REALM
+              value: NERC
+            - name: PROXY_AUTH_BASE_URL
+              value: https://keycloak.apps.obs.nerc.mghpcc.org
+            - name: PROXY_AUTH_TLS_VERIFY
+              value: 'true'
+            - name: PROXY_PROMETHEUS_BASE_URL
+              valueFrom:
+                secretKeyRef:
+                  key: PROXY_PROMETHEUS_BASE_URL
+                  name: prom-keycloak-proxy-secrets
+            - name: PROXY_PROMETHEUS_CA_CRT
+              value: /opt/prom-keycloak-proxy-certs/ca.crt
+            - name: PROXY_PROMETHEUS_TLS_CRT
+              value: /opt/prom-keycloak-proxy-certs/tls.crt
+            - name: PROXY_PROMETHEUS_TLS_KEY
+              value: /opt/prom-keycloak-proxy-certs/tls.key
+          image: quay.io/nerc-images/prom-keycloak-proxy:1.0.0
+          imagePullPolicy: Always
+          livenessProbe:
+            exec:
+              command:
+                - /usr/bin/curl
+                - http://localhost:8080
+            failureThreshold: 6
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          name: sso
+          ports:
+            - containerPort: 8080
+              name: http
+              protocol: TCP
+          readinessProbe:
+            exec:
+              command:
+                - /usr/bin/curl
+                - http://localhost:8080
```

Comment:
Ditto re: using an http probe

Reply:
Because all requests to this proxy need to be authenticated, the http liveness probes fail with:
`Readiness probe failed: HTTP probe failed with statuscode: 401`
Which is why I use curl instead, which doesn't complain about the status code.